src/hooks/useDebounce.js

```diff
@@ -0,0 +1,31 @@
+import { useCallback } from 'react';
+
+export const useDebounce = (func, waitAmount, immediate) => {
+  const debounceIt = () => {
+    // Closure timeout ref for all subsequent calls
+    let timeout;
+
+    // Return a func wrapper with the timeout var ref closure
+    return (...args) => {
+      const doItLater = () => {
+        timeout = null;
```

Comment:
If you use `clearTimeout` here does it do some weird stuff, since this is the function being called by the timeout?

Reply:
yea